src/wp-includes/blocks/navigation.php

```diff
@@ -241,11 +241,12 @@ private static function get_inner_blocks_from_navigation_post( $attributes ) {
 			// it encounters whitespace. This code strips it.
 			$blocks = block_core_navigation_filter_out_empty_blocks( $parsed_blocks );
 
-			// Run Block Hooks algorithm to inject hooked blocks.
-			$markup         = block_core_navigation_insert_hooked_blocks( $blocks, $navigation_post );
-			$root_nav_block = parse_blocks( $markup )[0];
-
-			$blocks = isset( $root_nav_block['innerBlocks'] ) ? $root_nav_block['innerBlocks'] : $blocks;
+			// Re-serialize, and run Block Hooks algorithm to inject hooked blocks.
+			// TODO: See if we can move the apply_block_hooks_to_content_from_post_object() call
+			// before the parse_blocks() call further above, to avoid the extra serialization/parsing.
+			$markup = serialize_blocks( $blocks );
+			$markup = apply_block_hooks_to_content_from_post_object( $markup, $navigation_post );
```

Comment:
Do we have a follow-up ticket where this (and the TODO below) are being tracked?

Reply:
cc @ockham